### PR TITLE
Add Breadcrumb documentation.

### PR DESCRIPTION
**Pull request by bretthoerner**
Still 😷, I hope this makes some sense.

### CODE REVIEW

File: docs/context.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
+Context & Breadcrumbs
+=====================
+
+The Java SDK implements the idea of a "context" to support attaching additional
+information to a specific user request, such as breadcrumbs.
+
+Unlike some other platforms, Java has no single definition of what "context" means.
```
Comment (tkaemming) [inline docs/context.rst]: The comparison to other platforms seems extraneous, and I also don't think this problem is particularly unique to Java. I'd reword this to clarify that the definition of context is dependent on your application (or framework) architecture and leave the rest out of it.

File: docs/context.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
+Context & Breadcrumbs
+=====================
+
+The Java SDK implements the idea of a "context" to support attaching additional
+information to a specific user request, such as breadcrumbs.
```
Comment (tkaemming) [inline docs/context.rst]: "specific user request" is a bit over-specific and limiting, it would be better to go with something more generic here.

File: docs/context.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+one thread per user request such as those based on synchronous servlet APIs. Raven
+also installs a ``ServletRequestListener`` that will clear the thread's context after
+each servlet request finishes.
+
+Raven defaults to the ``SingletonContextManager`` on Android, which maintains a single
+context instance for all threads for the lifetime of the application.
+
+As of version ``8.0.0`` to override the ``ContextManager`` you will need to override
+the ``getContextManager`` method in the ``DefaultRavenFactory``. A simpler API will
+be provided in the future.
```
Comment (tkaemming) [inline docs/context.rst]: `s/will/will likely/`

File: docs/context.rst [hunk 4]
```diff
@@ -16,12 +18,16 @@ Raven defaults to the ``SingletonContextManager`` on Android, which maintains a
...
 
 As of version ``8.0.0`` to override the ``ContextManager`` you will need to override
-the ``getContextManager`` method in the ``DefaultRavenFactory``. A simpler API will
+the ``getContextManager`` method in the ``DefaultRavenFactory``. A simpler API will likely
 be provided in the future.
 
 Using Breadcrumbs
 -----------------
 
+Breadcrumbs can be used to describe what was happening in an application leading
```
Comment (tkaemming) [inline docs/context.rst]: `s/what was happening/actions or events that occurred/` (… though I'm not sure if "event" is going to be confusing and overloaded here.)

File: docs/context.rst [hunk 5]
```diff
@@ -2,11 +2,13 @@ Context & Breadcrumbs
...
+information to events, such as breadcrumbs. A context may refer to a single
+request to a web framework, to the entire lifetime of an Android application,
+or something else that better suits your application's needs.
 
-Unlike some other platforms, Java has no single definition of what "context" means.
-For this reason a specific implementation must be chosen depending on what framework
-you are using. By default Raven uses a ``ThreadLocalContextManager`` that maintains
+There is no single definition of context that applies to every application,
+for this reason a specific implementation must be chosen depending on what your
+application does. By default Raven uses a ``ThreadLocalContextManager`` that maintains
```
Comment (tkaemming) [inline docs/context.rst]: This is less about what it does then how it is built, in my view — e.g. concurrency model (chosen by the application developer or framework author) will dictate the implementation in most cases.

File: docs/context.rst [hunk 6]
```diff
@@ -2,11 +2,13 @@ Context & Breadcrumbs
 =====================
 
 The Java SDK implements the idea of a "context" to support attaching additional
-information to a specific user request, such as breadcrumbs.
+information to events, such as breadcrumbs. A context may refer to a single
+request to a web framework, to the entire lifetime of an Android application,
+or something else that better suits your application's needs.
```
Comment (tkaemming) [inline docs/context.rst]: 👍

File: docs/context.rst (hunk 4 above)
Comment (bretthoerner) [inline docs/context.rst]: Yeah I think I prefer "actions that occurred" since `events` is pretty specific in the docs.

File: docs/context.rst (hunk 5 above)
Comment (bretthoerner) [inline docs/context.rst]: It's more meta than concurrency model, since Android uses `SingletonContextManager`... :(
